coffea/jetmet_tools/CorrectedMETFactory.py

```diff
@@ -125,27 +149,26 @@ def make_variant(*args):
         for unc in filter(
             lambda x: x.startswith(("JER", "JES")), awkward.fields(corrected_jets)
         ):
-            out[unc] = awkward.zip(
-                {
-                    "up": make_variant(
-                        MET[self.name_map["METpt"]],
-                        MET[self.name_map["METphi"]],
-                        corrected_jets[unc].up[self.name_map["JetPt"]],
-                        corrected_jets[unc].up[self.name_map["JetPhi"]],
-                        corrected_jets[unc].up[self.name_map["ptRaw"]],
-                    ),
-                    "down": make_variant(
-                        MET[self.name_map["METpt"]],
-                        MET[self.name_map["METphi"]],
-                        corrected_jets[unc].down[self.name_map["JetPt"]],
-                        corrected_jets[unc].down[self.name_map["JetPhi"]],
-                        corrected_jets[unc].down[self.name_map["ptRaw"]],
-                    ),
-                },
-                depth_limit=1,
-                with_name="METSystematic",
+            out_dict[unc] = awkward.virtual(
+                lazy_variant,
+                args=(
+                    unc,
+                    self.name_map["METpt"],
+                    self.name_map["METphi"],
+                    self.name_map["JetPt"],
+                    self.name_map["JetPhi"],
+                    self.name_map["ptRaw"],
+                    MET,
+                    corrected_jets,
+                ),
+                length=length,
```

Comment:
Just a note, since no `cache` is supplied here, we should avoid cache reference cycles.

Reply:
added non-user cache (fails the test with user cache so I suspect it is safe)